src/flows/components/FlowPipe.tsx

```diff
@@ -10,23 +9,22 @@ export interface FlowPipeProps {
 }
 
 const FlowPipe: FC<FlowPipeProps> = ({id}) => {
-  const panel: FC<PipeContextProps> = useMemo(
-    () => props => {
-      const _props = {
-        ...props,
-        id,
-      }
-
-      return createElement(FlowPanel, _props)
-    },
+  return useMemo(
+    () => (
+      <PipeProvider id={id}>
+        <Pipe
+          Context={props => {
+            const _props = {
+              ...props,
+              id,
+            }
+            return createElement(FlowPanel, _props)
+          }}
+        />
+      </PipeProvider>
```

Comment:
sanity check - why was the previous implementation being used and why the change?

Reply:
i think it's more of an oversight of when we added the "pass the container as a prop" idea. turns out that oversight cost us 15ms over a large number of results coming in